app/assets/javascripts/comments.js

```diff
@@ -1,151 +1,340 @@
-            commentEl.after("<li class='no-comments'><em>" + I18n.t("projects.index.no_comments") + "</em></li>");
-          }
+    var $form = $el.find('ul form');
+
+    $('.help-block', $form).addClass('hide');
+
+    $form.on('ajax:send', function () {
+      $('#comment_message', $form).attr('readonly', true);
+    })
+    .on('ajax:success', function (e, data) {
+      if (data.html) {
+        var list = $form.parents('ul');
+
+        // Remove potential 'no comments' element
+        list.parent().find('.content-comments')
+          .find('li.no-comments').remove();
+
+        list.parent().find('.content-comments')
+          .append('<li class="comment">' + data.html + '</li>')
+          .scrollTop(0);
+        list.parents('ul').find('> li.comment:gt(8)').remove();
+        $('#comment_message', $form).val('');
+        $('.form-group', $form)
+          .removeClass('has-error');
+        $('.help-block', $form)
+            .html('')
+            .addClass('hide');
+        scrollBottom($el);
+      }
+    })
+    .on('ajax:error', function (ev, xhr) {
+      if (xhr.status === 400) {
+        var messageError = xhr.responseJSON.errors.message;
 
-          // Case B
-          if (commentEl.prev(".comment-date-separator").length > 0 && commentEl.next(".comment").length == 0) {
-            commentEl.prev(".comment-date-separator").remove();
+        if (messageError) {
+          $('.form-group', $form)
+            .addClass('has-error');
+          $('.help-block', $form)
+              .html(messageError[0])
+              .removeClass('hide');
+        }
+      }
+    })
+    .on('ajax:complete', function () {
+      scrollBottom($('#comment_message', $form));
+      $('#comment_message', $form)
+        .attr('readonly', false)
+        .focus();
+    });
+  }
+
+  // restore comments after update or when new element is created
+  function bindCommentInitializerToNewElement() {
+    $(document)
+      .ready(function() {
+        if( document.getElementById('steps') !== null ) {
+          $('#steps')
+            .change(function() {
+              $('.step-save')
+                .on('click', function() {
+                  $(document)
+                    .on('ajax:success', function(){
+                      initializeComments();
+                    });
+                });
+            });
+        } else if ( document.getElementById('results') !== null ) {
+          $('#results')
+            .change(function() {
+              $('.save-result')
+                .on('click', function() {
+                  $(document)
+                    .on('ajax:success', function(){
+                      initializeComments();
+                    });
+                });
+            });
           }
-          commentEl.remove();
+    });
+  }
+
+  function initCommentOptions(scrollableContainer, useParentOffset) {
+    if ( ! _.isUndefined(typeof useParentOffset) ) {
```

Comment:
@mz3944 Or what? You gonna write me an angry message on email?! You gonna tell me mom? Boo-freaking-hoo! Look at me, I'm shaking. :fearful:

...*whisper* (please baby don't hurt me, don't hurt me, no more)